samples/tooling/profiles/profiles.cpp

```diff
@@ -0,0 +1,711 @@
+}
+
+Profiles::~Profiles()
+{
+	if (device)
+	{
+		// Clean up used Vulkan resources
+		// Note : Inherited destructor cleans up resources stored in base class
+
+		vkDestroyPipeline(get_device().get_handle(), pipeline, nullptr);
+
+		vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layout, nullptr);
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), base_descriptor_set_layout, nullptr);
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), sampler_descriptor_set_layout, nullptr);
+	}
+}
+
+void Profiles::generate_textures()
+{
+	// Generate random textures to be sourced from a single descriptor
+
+	// Image info is same for all textures
+	const int32_t dim = 2;
+
+	VkImageCreateInfo image_info = vkb::initializers::image_create_info();
+	image_info.format            = VK_FORMAT_R8G8B8A8_UNORM;
+	image_info.extent            = {dim, dim, 1};
+	image_info.mipLevels         = 1;
+	image_info.arrayLayers       = 1;
+	image_info.initialLayout     = VK_IMAGE_LAYOUT_UNDEFINED;
+	image_info.usage             = VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT;
+	image_info.imageType         = VK_IMAGE_TYPE_2D;
+	image_info.samples           = VK_SAMPLE_COUNT_1_BIT;
+	image_info.tiling            = VK_IMAGE_TILING_OPTIMAL;
+	image_info.sharingMode       = VK_SHARING_MODE_EXCLUSIVE;
+
+	VkImageViewCreateInfo image_view           = vkb::initializers::image_view_create_info();
+	image_view.viewType                        = VK_IMAGE_VIEW_TYPE_2D;
+	image_view.format                          = VK_FORMAT_R8G8B8A8_UNORM;
+	image_view.subresourceRange.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
+	image_view.subresourceRange.baseMipLevel   = 0;
+	image_view.subresourceRange.levelCount     = 1;
+	image_view.subresourceRange.baseArrayLayer = 0;
+	image_view.subresourceRange.layerCount     = 1;
+
+	auto staging_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                          image_info.extent.width * image_info.extent.height * sizeof(uint32_t),
+	                                                          VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
+	                                                          VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	textures.resize(32);
+	for (size_t i = 0; i < textures.size(); i++)
+	{
+		VK_CHECK(vkCreateImage(get_device().get_handle(), &image_info, nullptr, &textures[i].image));
+		VkMemoryAllocateInfo memory_allocation_info = vkb::initializers::memory_allocate_info();
+		VkMemoryRequirements memory_requirements;
+		vkGetImageMemoryRequirements(get_device().get_handle(), textures[i].image, &memory_requirements);
+		memory_allocation_info.allocationSize  = memory_requirements.size;
+		memory_allocation_info.memoryTypeIndex = get_device().get_memory_type(memory_requirements.memoryTypeBits, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+		VK_CHECK(vkAllocateMemory(get_device().get_handle(), &memory_allocation_info, nullptr, &textures[i].memory));
```

Comment:
Not important, but why not using VMA ?

Reply:
Good question. I guess I just didn't think about using VMA. I may take a look at this after the sample is merged.